packages/zowe-explorer/src/job/ZosJobsProvider.ts

```diff
@@ -878,21 +880,24 @@ export class ZosJobsProvider extends ZoweTreeProvider implements IZoweTree<IZowe
                 this.resetJobProperties(jobProperties);
                 return Promise.resolve(searchCriteriaObj);
             default:
-                const options: vscode.InputBoxOptions = {
-                    value: jobProperties.find((prop) => prop.label === pattern).value,
-                    placeHolder: jobProperties.find((prop) => prop.label === pattern).placeHolder,
-                    validateInput: (text) => {
-                        return jobProperties.find((prop) => prop.label === pattern)?.validateInput(text);
-                    },
-                };
-                jobProperties.find((prop) => prop.label === pattern).value = await Gui.showInputBox(options);
+                let property = jobProperties.find((prop) => prop.label === pattern);
+                if (property != null) {
```

Comment:
What if property is undefined?

Reply:
`!= null` is not a strict type check, so since `null == undefined`, `!= null` will account for both

Reply:
I think this will lead to confusion and linters recommend using strict checking to avoid this confusion  `===`

From JavaScript the good parts book: 
`JavaScript has two sets of equality operators: === and !==, and their evil twins == and !=. The good ones work the way you would expect. If the two operands are of the same type and have the same value, then === produces true and !== produces false. The evil twins do the right thing when the operands are of the same type, but if they are of different types, they attempt to coerce the values. the rules by which they do that are complicated and unmemorable.`

Looks like this function is checking if property is defined, then go head and assign values. Perhaps it would be better to check for truth-ish value `if (property)` or something like that.

Reply:
While I generally agree with the excerpt from the JS book, I'm unaware of an existent JavaScript runtime where `undefined` is coerced to a value that's not equal to `null`. If others agree that we should be using `===` here, then I will definitely update this check. For more context, we are already adopting this convention in other areas of the codebase, so I'm merely maintaining the same trend for consistency. 

For the suggestion on `if (property)`: AFAIK JavaScript will implicitly coerce the `property` value to `true/false` if used in this context, so type coercion would still be done here as well. The main difference would be an implicit coercion rather than explicit. However, I'm open to suggestions and willing to discuss this in further detail during stand-up if desired 🙂  

If we would want to change this to strict equality, we should probably update other areas of the codebase (where this check is being used) to maintain consistency.

Thoughts @zFernand0 @t1m0thyj ?

Reply:
Yes, JavaScript will still coerce `if (property)`, however, this check will return false for any `falsy `condition: https://developer.mozilla.org/en-US/docs/Glossary/Falsy

What if property is `false` or an empty string? The logic inside the if statement should not execute if `property` is any falsy value. It's safer to check for any any falsy value.

Reply:
> What if `property` is `false` or an empty string? 

It will never be `false` or an empty string though - in order to compile any calls to `handleEditingMultiJobParameters` w/ TypeScript, `jobProperties` must be of type `IJobPickerOption[]`:

```ts
interface IJobPickerOption {
    key: string;
    label: string;
    value: string;
    show: boolean;
    placeHolder: string;
    validateInput?;
}
```

If calls were made to this function that don't follow the type signature, I would argue that the function is being improperly used, as the type system in TypeScript is supposed to prevent incorrect calls from being compiled by design. However, I understand the concern and am willing to update this, just wanted to get some more perspectives on this topic before continuing.

Reply:
Yes, using TypeScript will make it less likely that wrong type will be passed, however, in the future somebody could re-factor the code and eliminate the interface or bypass type checking or use the function improperly. If we have more robust check here it will lead to less bugs.

Reply:
This is an interesting topic 😋 
I can see both side of the argument.

I'm curious why `eliminating an interface` won't be a red-flag for code-reviewers? 

I can understand that there may be places (mostly in tests) where we bypass certain types, but I'm sure that anyone reviewing a PR with an `any` in the codebase (not tests) may have some questions at least.

For what is worth, I believe the CLI squad has adopted the practice of using `!= null` and `!== <anything-else>`.

I understand that there are instances where it makes sense to check for falsy values. but I agree with Trae's point that this very specific case should not be called with a falsy value.

I'm personally biased since the language (JS/TS) allows us to type less with `!=`. However, I'm also ok if the squad decides to change their opinion and always using strict in-/equality going forward.